pkg/apis/networkservicemesh.io/v1/types.go

```diff
@@ -24,17 +26,34 @@ import (
 const (
 	NSMGroup           = "networkservicemesh.io"
 	NSMGroupVersion    = "v1"
-	NSMEPPlural        = "networkserviceendpoints"
-	FullNSMEPName      = NSMEPPlural + "." + NSMGroup
-	NSMChannelPlural   = "networkservicechannels"
-	FullNSMChannelName = NSMChannelPlural + "." + NSMGroup
-	NSMPlural          = "networkservices"
-	FullNSMName        = NSMPlural + "." + NSMGroup
+	NSMSpecPath        = "github.com/ligato/networkservicemesh/pkg/apis/" + NSMGroup
+	NSMEPSingular      = "networkserviceendpoint"
+	NSMEPPlural        = NSMEPSingular + "s"
+	NSMChannelSingular = "networkservicechannel"
+	NSMChannelPlural   = NSMChannelSingular + "s"
+	NSMSingular        = "networkservice"
+	NSMPlural          = NSMSingular + "s"
+)
+
+var (
+	// NSMEPTypeName is the reflected type name of a NetworkServiceEndpoint
+	NSMEPTypeName = reflect.TypeOf(NetworkServiceEndpoint{}).Name()
+	// FullNSMEPName is the full name (including path) of NetworkServiceEndpoints
+	FullNSMEPName = NSMSpecPath + "/" + NSMGroupVersion + "." + NSMEPTypeName
+	// NSMChannelTypeName is the reflected type name of a NetworkServiceChannel
+	NSMChannelTypeName = reflect.TypeOf(NetworkServiceChannel{}).Name()
+	// FullNSMChannelName is the full name (including path) of NetworkServiceChannels
+	FullNSMChannelName = NSMSpecPath + "/" + NSMGroupVersion + "." + NSMChannelTypeName
+	// NSMTypeName is the reflected type name of a NetworkService
+	NSMTypeName = reflect.TypeOf(NetworkService{}).Name()
+	// FullNSMName is the full name (including path) of NetworkServices
+	FullNSMName = NSMSpecPath + "/" + NSMGroupVersion + "." + NSMTypeName
 )
 
 // NetworkServiceEndpoint CRD
 // +genclient
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
```

Comment:
QQ: Since it looks like the first line of the comment is being taken from the proto file, might it be possible to roll the k8s: annotations into those comments so they get automatically carried over?

Reply:
These are per-structure dependent. For example, for the List structures, we do not generate clients. I may not be following the comment however, so please let me know if I've misunderstood. :)